src/test/java/org/sagebionetworks/bridge/spring/controllers/ScheduledActivityControllerTest.java

```diff
@@ -0,0 +1,568 @@
+        assertTrue(STARTS_ON.isEqual(startsOnCaptor.getValue()));
+        assertTrue(ENDS_ON.isEqual(endsOnCaptor.getValue()));
+    }
+
+    @Test
+    public void updateScheduledActivitiesWithClientData() throws Exception {
+        JsonNode clientData = TestUtils.getClientData();
+        
+        DynamoScheduledActivity schActivity = new DynamoScheduledActivity();
+        schActivity.setTimeZone(DateTimeZone.UTC);
+        schActivity.setGuid(BridgeUtils.generateGuid());
+        schActivity.setLocalScheduledOn(LocalDateTime.now().minusDays(1));
+        schActivity.setActivity(ACTIVITY_3);
+        schActivity.setClientData(clientData);
+        
+        TestUtils.mockRequestBody(mockRequest, ImmutableList.of(schActivity));
+        
+        StatusMessage result = controller.updateScheduledActivities();
+        assertEquals(result, ScheduledActivityController.UPDATED_MSG);
+        
+        verify(mockScheduledActivityService).updateScheduledActivities(eq(HEALTH_CODE), activitiesCaptor.capture());
+        
+        List<ScheduledActivity> capturedActivities = activitiesCaptor.getValue();
+        assertEquals(capturedActivities.get(0).getClientData(), clientData);
+    }
+
+    @Test
+    public void getScheduledActivitiesV4() throws Exception {
+        DateTimeZone zone = DateTimeZone.forOffsetHours(4);
+        DateTime startsOn = DateTime.now(zone).minusMinutes(1);
+        DateTime endsOn = DateTime.now(zone).plusDays(7);
+        
+        String result = controller.getScheduledActivitiesByDateRange(startsOn.toString(), endsOn.toString());
+        
+        JsonNode node = BridgeObjectMapper.get().readTree(result);
+        assertEquals(node.get("startTime").textValue(), startsOn.toString());
+        assertEquals(node.get("endTime").textValue(), endsOn.toString());
+        
+        verify(sessionUpdateService).updateTimeZone(any(UserSession.class), timeZoneCaptor.capture());
+        verify(mockCacheProvider).updateRequestInfo(requestInfoCaptor.capture());
+        verify(mockScheduledActivityService).getScheduledActivitiesV4(eq(STUDY), contextCaptor.capture());
+        
+        assertEquals(startsOn.getZone(), timeZoneCaptor.getValue());
+        
+        RequestInfo requestInfo = requestInfoCaptor.getValue();
+        assertEquals(requestInfo.getUserId(), "id");
+        assertEquals(requestInfo.getClientInfo(), CLIENT_INFO);
+        assertEquals(requestInfo.getUserAgent(), USER_AGENT);
+        assertEquals(requestInfo.getLanguages(), LANGUAGES);
+        assertEquals(requestInfo.getUserDataGroups(), USER_DATA_GROUPS);
+        assertEquals(requestInfo.getUserSubstudyIds(), USER_SUBSTUDY_IDS);
+        assertTrue(requestInfo.getActivitiesAccessedOn().isAfter(startsOn));
+        assertNull(requestInfo.getSignedInOn());
+        assertEquals(requestInfo.getStudyIdentifier(), TEST_STUDY);
+        verify(controller).persistTimeZone(session, zone);
```

Comment:
RequestInfo.getTimeZone() doesn't change from its default value which is UTC. It's poorly named (it's the initial time zone of the user and that hasn't been set). It's about to be MSK but without the SessionUpdateService to actually update the session before creating RequestInfo, it's not in the test. Added a verify that the mock account is being updated with the time zone.

Not sure how else to handle this since it was incorrect to use the SessionUpdateService implementation instead of mocking it. the call to the service is being made correctly and that's tested separately.